### PR TITLE
Invalid attribute values in the `<NLog>` element should not break logging & better attribute parsing order

### LINKED ISSUE #2930: No logs written from .net core console app
Demo project here: [NLogTest](https://github.com/fschick/NLogTest)

**Very intresting detail**: As soon as the `internalLogLevel` is set to other than `None` e.g. `internalLogLevel="Trace"`, it's all workging without any noticeable error/warning/debug message.

**NLog version**: 4.5.10
**NLog.Extensions.Logging**: 1.3.0
**Platform**: .NET Core 2

**Current NLog config**
```<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" 
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
      autoReload="true" 
      throwExceptions="true" 
      internalLogFile="C:\NLog.internal.log" 
      internalLogLevel="None">
  <targets>
    <target name="logfile"
            xsi:type="File"
            fileName="${basedir}\NLogTest.log"
            layout="${date}|${level}|${message}|">
    </target>
  </targets>
  <rules>
    <logger name="*" minlevel="Trace" writeTo="logfile" />
  </rules>
</nlog>
```
**Current code**
```
private static void Main(string[] args)
{
    using (var services = CreateDiServiceProvider())
    {
        var logger = services.GetRequiredService<ILogger<Program>>();
        logger.LogCritical("Test");
    }

    LogManager.Shutdown();
}

private static ServiceProvider CreateDiServiceProvider()
{
    IServiceCollection services = new ServiceCollection();
    services.AddLogging(ConfigureLogging);
    return services.BuildServiceProvider();
}

private static void ConfigureLogging(ILoggingBuilder loggingBuilder)
{
    LogManager.LoadConfiguration("NLogTest.nlog");
    loggingBuilder.SetMinimumLevel(LogLevel.Trace);
    loggingBuilder.AddNLog(new NLogProviderOptions { CaptureMessageTemplates = true, CaptureMessageProperties = true });
}
```

- What is the current result: *No logs are written to log file*
- What is the expected result? *Logs should be written log file*
- Did you checked the Internal log: *yes*
- Please post full exception details: *no messages, no exceptions*
- Are there any workarounds? *yes, set `internalLogLevel` to other than `None`*
- Is there a version in which it did work? *no*
- Can you help us by writing an unit test? *unsure, test project provided*

As mentioned above, with the given configuration no logs are written. Demo project here: [NLogTest](https://github.com/fschick/NLogTest)

### PR DESCRIPTION
**Pull request by 304NotModified**
+dead code removal

fixes #2930

### CODE REVIEW

**Comment by 304NotModified** [comment]
oops forgot about this one: #2891

**Comment by snakefoot** [comment]
> oops forgot about this one: #2891

Thank you for remembering

File: src/NLog/Config/LoggingConfigurationParser.cs [hunk 1]
```diff
@@ -659,7 +714,7 @@ private void ParseTargetElement(Target target, ILoggingConfigurationElement targ
         }
 
         private bool ParseTargetWrapper(Dictionary<string, ILoggingConfigurationElement> typeNameToDefaultTargetParameters, string name, ILoggingConfigurationElement childElement,
```
Comment (NLog-SonarQube) [inline src/NLog/Config/LoggingConfigurationParser.cs]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)

**Comment by NLog-SonarQube** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L71](https://github.com/NLog/NLog/blob/79c8751f592e38148372c2f6292eaa679159b9b3/src/NLog/Config/LoggingConfigurationParser.cs#L71): Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L230](https://github.com/NLog/NLog/blob/79c8751f592e38148372c2f6292eaa679159b9b3/src/NLog/Config/LoggingConfigurationParser.cs#L230): Refactor this method to reduce its Cognitive Complexity from 27 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L464](https://github.com/NLog/NLog/blob/79c8751f592e38148372c2f6292eaa679159b9b3/src/NLog/Config/LoggingConfigurationParser.cs#L464): Refactor this method to reduce its Cognitive Complexity from 41 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L608](https://github.com/NLog/NLog/blob/79c8751f592e38148372c2f6292eaa679159b9b3/src/NLog/Config/LoggingConfigurationParser.cs#L608): Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L759](https://github.com/NLog/NLog/blob/79c8751f592e38148372c2f6292eaa679159b9b3/src/NLog/Config/LoggingConfigurationParser.cs#L759): Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![MINOR][MINOR] [LoggingConfigurationParser.cs#L854](https://github.com/NLog/NLog/blob/79c8751f592e38148372c2f6292eaa679159b9b3/src/NLog/Config/LoggingConfigurationParser.cs#L854): Remove this redundant jump. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3626)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/NLog/Config/LoggingConfigurationParser.cs [hunk 2]
```diff
@@ -658,12 +786,15 @@ private void ParseTargetElement(Target target, ILoggingConfigurationElement targ
             }
         }
 
-        private bool ParseTargetWrapper(Dictionary<string, ILoggingConfigurationElement> typeNameToDefaultTargetParameters, string name, ILoggingConfigurationElement childElement,
-    WrapperTargetBase wrapper)
+        private bool ParseTargetWrapper(
```
Comment (NLog-SonarQube) [inline src/NLog/Config/LoggingConfigurationParser.cs]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)

File: src/NLog/Config/LoggingConfigurationParser.cs [hunk 3]
```diff
@@ -698,10 +830,13 @@ private void ParseTargetElement(Target target, ILoggingConfigurationElement targ
 
                 return true;
             }
+
             return false;
         }
 
-        private bool ParseCompoundTarget(Dictionary<string, ILoggingConfigurationElement> typeNameToDefaultTargetParameters, string name, ILoggingConfigurationElement childElement,
+        private bool ParseCompoundTarget(
```
Comment (NLog-SonarQube) [inline src/NLog/Config/LoggingConfigurationParser.cs]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)

File: src/NLog/Config/LoggingConfigurationParser.cs [hunk 4]
```diff
@@ -168,10 +256,13 @@ protected virtual bool ParseNLogSection(ILoggingConfigurationElement configSecti
...
             }
+
             return false;
         }
 
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom", Justification = "Need to load external assembly.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability",
+            "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom",
+            Justification = "Need to load external assembly.")]
         private void ParseExtensionsElement(ILoggingConfigurationElement extensionsElement, string baseDirectory)
```
Comment (NLog-SonarQube) [inline src/NLog/Config/LoggingConfigurationParser.cs]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 27 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)

**Comment by NLog-SonarQube** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L71](https://github.com/NLog/NLog/blob/b23cf0618eee871d2361b13fa10bf273add07e92/src/NLog/Config/LoggingConfigurationParser.cs#L71): Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L505](https://github.com/NLog/NLog/blob/b23cf0618eee871d2361b13fa10bf273add07e92/src/NLog/Config/LoggingConfigurationParser.cs#L505): Refactor this method to reduce its Cognitive Complexity from 41 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![CRITICAL][CRITICAL] [LoggingConfigurationParser.cs#L670](https://github.com/NLog/NLog/blob/b23cf0618eee871d2361b13fa10bf273add07e92/src/NLog/Config/LoggingConfigurationParser.cs#L670): Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)
1. ![MINOR][MINOR] [LoggingConfigurationParser.cs#L940](https://github.com/NLog/NLog/blob/b23cf0618eee871d2361b13fa10bf273add07e92/src/NLog/Config/LoggingConfigurationParser.cs#L940): Remove this redundant jump. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3626)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by 304NotModified** [comment]
@snakefoot any opinion on this? If not, I like to merge it

**Comment by snakefoot** [comment]
Looks good to me

**Comment by 304NotModified** [comment]
Thanks :)

**Pull request merged**